src/main/java/edu/ucsb/cs156/gauchoride/controllers/DriverAvailabilityController.java

```diff
@@ -120,6 +130,10 @@ public Object deleteDriverAvailability(
         DriverAvailability availability = driverAvailabilityRepository.findById(id)
                 .orElseThrow(() -> new EntityNotFoundException(DriverAvailability.class, id));
 
+        if (availability.getDriverId() != getCurrentUser().getUser().getId()) {
+            throw new EntityNotFoundException(DriverAvailability.class, id);
```

Comment:
This is not the correct exception to throw.  This should only be used when you did a search for a DriverAvailability with a certain id and it was not found.  Throwing this exception will cause future devs to curse your name.

The exception you want is one that will turn into a 403 unauthorized, but if you can't figure out how to do that, then just throw a generic exception with a reasonable error message.